rel-eng/custom/custom.py

```diff
@@ -195,6 +197,22 @@ def _fetch_local(self):
 
         return gitrev
 
+    def _fetch_prebuilt_local(self):
+        source_files = [os.path.expanduser(source_file) for source_file in self.builder.args['source_files']]
+        dest_dir = os.path.join(self.builder.rpmbuild_sourcedir, 'archive')
+        if not os.path.exists(dest_dir):
+            os.mkdir(dest_dir)
+
+        for src in source_files:
+            debug("Copying %s to %s" % (src, os.path.join(dest_dir, os.path.basename(src))))
+            shutil.copy(src, os.path.join(dest_dir, os.path.basename(src)))
+
+        gitsha = self.builder.args['git_hash'][0]
```

Comment:
This will raise an `IndexError` or a `KeyError` if it's not passed in. IMHO that's not a problem and actually a feature, but I'd do it at the start of the function so we reject builds without a git sha. We can also raise an exception with a bit more helpful error message.